chart/mqtt-connector/templates/deployment.yml

```diff
@@ -0,0 +1,104 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    # Original Helm labels v
+    app: {{ template "connector.fullname" . }}
+    component: mqtt-connector
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    # Helm/k8s recommended label names v
+    app.kubernetes.io/name: {{ template "connector.fullname" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/version: {{ .Chart.Version }}
+    app.kubernetes.io/component: mqtt-connector
+    app.kubernetes.io/part-of: openfaas
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+  name: {{ template "connector.fullname" . }}
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      app: {{ template "connector.fullname" . }}
+      component: mqtt-connector
+
+  template:
+    metadata:
+      annotations:
+        prometheus.io.scrape: "false"
+      labels:
+        app: {{ template "connector.fullname" . }}
+        component: mqtt-connector
+    spec:
+      {{- if .Values.basic_auth }}
+      volumes:
+        - name: auth
+          secret:
+            secretName: basic-auth
+      {{- end }}
+      containers:
+        - name: connector
+          image: {{ .Values.image }}
+          command:
+          - "/usr/bin/connector"
+          args:
+          {{- if .Values.broker }}
```

Comment:
does this work even if broker is empty? or if we set `-broker=""` does the connector still work, if so we could reduce the templating noise here by removing the if blocks

Reply:
It can because this is flag for the connector which has default value, so if nothing is provided to the flag it will use default value.